mahjongiOS/Controller/ViewController.swift

```diff
@@ -92,31 +94,41 @@ class ViewController: UIViewController, GameDelegate {
          tile selected during discard, call/expose and joker trade to be
          decoupled.
         */
-        let button = sender as! UIButton
+        exchangeButton.isEnabled = false
+        discardButton.isEnabled = false
         let tileIndex = button.tag
-        if discardState && selectedIndexes.count > 0 && !button.isSelected {
-            let alert = UIAlertController(title: "Error", message: "Select single tile to discard", preferredStyle: .alert)
-            alert.addAction(UIAlertAction(title: NSLocalizedString("OK", comment: "Default action"), style: .default))
-            self.present(alert, animated: true, completion: nil)
-            return
+        if discardState && selectedIndexes.count == 1 && !button.isSelected {
+            let previous_selection = tileButtons[selectedIndexes[0]]
+            if button.isExposed != previous_selection.isExposed {
```

Comment:
This is for when you want to swap, right? Took me awhile to figure that out (but then again I haven't looked at this code in awhile). You could add a variable like `isSwapping`, just to make it a bit more clear. Ex. something like:
```swift
let isSwapping = button.isExposed != previous_selection.isExposed 
```

Could also write out the longer form of that condition:
```swift
let isSwapping = (button.isExposed && !previous_selection.isExposed) || (!button.isExposed && previous_selection.isExposed)
```

But these 2 suggestions make the code longer and the current code is still quite readable, so feel free to ignore!